src/hotspot/share/nmt/vmatree.hpp

```diff
@@ -53,16 +54,18 @@ class VMATree {
     }
   };
 
-  enum class StateType : uint8_t { Reserved, Committed, Released, LAST };
+  // Bit fields view: bit 0 for Reserved, bit 1 for Committed.
+  // Setting a region as Committed preserves the Reserved state.
+  enum class StateType : uint8_t { Reserved = 1, Committed = 3, Released = 0, COUNT = 4 };
```

Comment:
As the comments above it says, and we already had a discussion on it: The 'Committed' state now means 'Reserved and Committed'. So, when we visit nodes and check for Reserved ones, the committed nodes are also considered as Reserved. Otherwise, we would ignore the Committed nodes and continue looking for Reserved nodes.

Reply:
Right, I don't remember the discussion on this being a bit field. I agree, though, Committed also means Reserved, but it's not necessary to express that as a bit field. Regardless, I leave it up to you how we express this.